### PR TITLE
Add a ClassLoader parameter to resolveTracer

### LINKED ISSUE #37: Need a way to pass in a ClassLoader to ServiceLoader
When using `TracerResolver` class in a server environment where there are more than one class loader, there is no way to pass in a particular `ClassLoader` object for the `ServiceLoader`.

I am proposing to add a new method in `TracerResolver`:

```
     public static Tracer resolveTracer(ClassLoader classloader) {
    	if (classloader == null) {
    		classloader = Thread.currentThread().getContextClassLoader();
    	}
        try { // Take care NOT to import GlobalTracer as it is an optional dependency and may not be on the classpath.
            if (io.opentracing.util.GlobalTracer.isRegistered()) {
                return logResolved(io.opentracing.util.GlobalTracer.get());
            }
        } catch (NoClassDefFoundError globalTracerNotInClasspath) {
            LOGGER.finest("GlobalTracer is not found on the classpath.");
        }

        Tracer tracer = null;
        if (!TracerResolver.isDisabled()) {
            tracer = getFromFactory(classloader);
            if (null == tracer) {
                tracer = getFromResolver(classloader);
            }

            if (null == tracer) {
                tracer = getFromServiceLoader(classloader);
            }
        }

        return tracer;
    }
``` 

and `getFromFactory(classloader)` can look like this:

```
    private static Tracer getFromFactory(ClassLoader classloader) {
        for (TracerFactory factory : prioritize(ServiceLoader.load(TracerFactory.class, classloader))) {
            try {
                Tracer tracer = convert(factory.getTracer());
                if (tracer != null) {
                    return logResolved(tracer);
                }
            } catch (RuntimeException rte) {
                LOGGER.log(Level.WARNING, "Error getting tracer using " + factory + ": " + rte.getMessage(), rte);
            }
        }

        return null;
    }
```

Now, I can pass in a `ClassLoader` object where it can find and load the implementation classes (e.g. `JaegerTracerFactory`).

Comments?

### PR DESCRIPTION
**Pull request by fmhwong**
Fixes #37 

Add a `ClassLoader` parameter to `TracerResolver.resolveTracer()` method so that it allows to use an alternate `ClassLoader`.

### CODE REVIEW

File: opentracing-tracerresolver/src/main/java/io/opentracing/contrib/tracerresolver/TracerResolver.java [hunk 1]
```diff
@@ -83,15 +115,18 @@ public static Tracer resolveTracer() {
             LOGGER.finest("GlobalTracer is not found on the classpath.");
         }
 
+        if (classloader == null) {
```
Comment (sjoerdtalsma) [inline opentracing-tracerresolver/src/main/java/io/opentracing/contrib/tracerresolver/TracerResolver.java]: Could you move this inside the `!isDisabled` branch? Then there is no unnecessary call done if the resolution is actually disabled.

**Review by sjoerdtalsma — CHANGES_REQUESTED** [review]
Minor comment, for the rest the code looks ok at first glance

File: opentracing-tracerresolver/src/main/java/io/opentracing/contrib/tracerresolver/TracerResolver.java (hunk 1 above)
Comment (fmhwong) [reply opentracing-tracerresolver/src/main/java/io/opentracing/contrib/tracerresolver/TracerResolver.java]: Resolved. Thanks!

File: opentracing-tracerresolver/src/main/java/io/opentracing/contrib/tracerresolver/TracerResolver.java [hunk 2]
```diff
@@ -85,13 +117,16 @@ public static Tracer resolveTracer() {
 
         Tracer tracer = null;
         if (!TracerResolver.isDisabled()) {
-            tracer = getFromFactory();
+            if (classloader == null) {
+        		classloader = Thread.currentThread().getContextClassLoader();
```
Comment (sjoerdtalsma) [inline opentracing-tracerresolver/src/main/java/io/opentracing/contrib/tracerresolver/TracerResolver.java]: Nit: indentation looks 'off'; could it be spaces vs tabs or did you add so many spaces intentionally?

Comment (fmhwong) [reply opentracing-tracerresolver/src/main/java/io/opentracing/contrib/tracerresolver/TracerResolver.java]: Good catch!  I used a new Eclipse workspace and forgot to change the default tab key to space.

**Review by sjoerdtalsma — APPROVED** [review]

**Comment by sjoerdtalsma** [comment]
Thanks @fmhwong for this contribution!

**Pull request merged**